test/org/zaproxy/zap/extension/pscanrulesAlpha/XPoweredByHeaderInfoLeakScannerUnitTest.java

```diff
@@ -0,0 +1,95 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import org.junit.Test;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+/**
+ * @author Vahid Rafiei (@vahid_r)
+ */
+
+public class XPoweredByHeaderInfoLeakScannerUnitTest extends PassiveScannerTest {
+
+    @Override
+    protected PluginPassiveScanner createScanner() {
+        return new XPoweredByHeaderInfoLeakScanner();
+    }
+
+    @Test
+    public void shouldNotRaiseAlertIfThereIsNoXPoweredBy() throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                        "Server: Apache-Coyote/1.1\r\n");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void shouldRaiseAnAlertIfFindsXPoweredBy() throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                        "Server: Apache-Coyote/1.1\r\n" +
+                        "X-Powered-By: Servlet/3.0\r\n");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), is(1));
+        assertThat(alertsRaised.get(0).getEvidence(), is("X-Powered-By"));
+    }
+
+    @Test
+    public void shouldRaiseMultipleAlersWithOnlyOneEvidencetIfFindsMultipleXPoweredBy() throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                        "Server: Apache-Coyote/1.1\r\n" +
+                        "X-Powered-By: PHP/5.4\r\n" +
+                        "X-Powered-By: Servlet/3.0\r\n" +
+                        "X-Powered-By: ASP.NET\r\n");
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        // Then
+        assertThat(alertsRaised.size(), is(3));
+        assertThat(alertsRaised.get(0).getEvidence(), is("X-Powered-By"));
+        assertThat(alertsRaised.get(1).getOtherInfo(), is("X-Powered-By"));
```

Comment:
It should raise just one alert, subsequent (or all?)  `X-Powered-By` headers (i.e. its values) would be added to the other info of the ("first") alert.